### ISSUE TITLE
Challenge 1 - Rahul Kumar#3373

### ISSUE THREAD

**Issue by iamrahul8** [scaleracademy/scaler-september-open-source-challenge#105]
name: Rahul Kumar
github_user_name: iamrahul8

**Comment by iamrahul8** [comment]
**URL of your forked repository :** https://github.com/iamrahul8/scaler-september-open-source-challenge

## A screenshot of the cloned project in my system
![image](https://user-images.githubusercontent.com/72541381/188096776-b27c4851-1d94-4c61-9de4-532f29cd510d.png)

**Comment by iamrahul8** [comment]
Day 3
# Screenshot of branches : 
![image](https://user-images.githubusercontent.com/72541381/188261097-4ad5fa39-da54-498f-9bf0-49939af15931.png)

![image](https://user-images.githubusercontent.com/72541381/188261094-7b7517ca-7106-4386-821d-7b6486f8cbba.png)

**Comment by iamrahul8** [comment]
Day 4
# Screenshot showing the latest repository contents
![image](https://user-images.githubusercontent.com/72541381/188304975-f05e901b-7dd9-46fa-8f83-af2af00b948c.png)

![image](https://user-images.githubusercontent.com/72541381/188304969-425754c6-94fd-411b-b3a1-e2c5e4fc6ab3.png)

**Comment by iamrahul8** [comment]
# Challenge - 9
Successfully removed commit created in challenge 8
![image](https://user-images.githubusercontent.com/72541381/189345941-9388455a-5135-4fef-8d05-24028031c30d.png)


![image](https://user-images.githubusercontent.com/72541381/189345275-7151fc91-2b2d-42e3-9795-688416b23430.png)

**Comment by iamrahul8** [comment]
# Challenge 11
Created new branch
![image](https://user-images.githubusercontent.com/72541381/189525747-f14af25d-110f-49ed-bdfd-f73ca2ae2f98.png)

**Comment by iamrahul8** [comment]
# Challenge 13
![image](https://user-images.githubusercontent.com/72541381/189877910-d3706ce2-8313-4949-84f7-88297ca82dea.png)

**Comment by iamrahul8** [comment]
# Challenge 14
Resolving merge conflicts
![Screenshot (2291)](https://user-images.githubusercontent.com/72541381/190219316-c0c47df7-04df-4f56-8926-93571dc25f40.png)
![Screenshot (2294)](https://user-images.githubusercontent.com/72541381/190219691-1b1ad722-2bfd-4b2d-94bf-b75f0b5bccd2.png)

![Screenshot (2293)](https://user-images.githubusercontent.com/72541381/190219719-927bf45e-2b82-4f70-8024-cc10ecd3b647.png)

**Comment by iamrahul8** [comment]
# Challenge 16 
completed fixed Merge conflict
![image](https://user-images.githubusercontent.com/72541381/190620458-35b5d7d8-9513-4b62-88ee-f5aa5c9d0337.png)
![image](https://user-images.githubusercontent.com/72541381/190620520-0643a2a3-e013-4f6f-8b1c-c6e187b08e7a.png)
![image](https://user-images.githubusercontent.com/72541381/190620741-334e4a99-8db7-4bf2-b0b3-6ecb6f59c96c.png)

**Comment by iamrahul8** [comment]
# Challenge 17
repo link : https://github.com/iamrahul8/Scaler-Challenge-17
---
|  Created repo   |  Using GitHub Desktop  |
|---|---|
| ![image](https://user-images.githubusercontent.com/72541381/190864641-b69109c3-8c69-4669-897a-254ae4b7e561.png)   | ![image](https://user-images.githubusercontent.com/72541381/190864659-b7f18ccf-795f-46bd-9f88-60b0089d6fb5.png)  |

 ![image](https://user-images.githubusercontent.com/72541381/190864662-c8f0042e-53a9-45d5-969c-b3fde03fde2a.png)

Repo ss -
![image](https://user-images.githubusercontent.com/72541381/190864669-34ddf1b8-299e-44d3-9977-ab694b0d4c77.png)

**Comment by iamrahul8** [comment]
# Challenge 21 - Done ✅
---
Aliases created :

![image](https://user-images.githubusercontent.com/72541381/191528799-f0118eb6-33ca-4ab7-8abc-808d108761c4.png)

Using aliases:

![image](https://user-images.githubusercontent.com/72541381/191528821-ebd8753d-3487-4726-b9c0-9bbbdab505c6.png)

Uses in another repo :
![image](https://user-images.githubusercontent.com/72541381/191529688-f5924cd3-55f8-4136-a20d-8df535191d9c.png)

**Comment by kuraakhilesh8230** [comment]
Congrats you have successfully completed milestone 2 and you are good to move onto next milestone
All the best✅️

**Comment by iamrahul8** [comment]
# Challenge 23 ✅
[![forthebadge](https://forthebadge.com/images/badges/open-source.svg)](https://scaler.com)
>What needs to be done for effective open source contribution, especially with respect to open source issue creation, best practices around branch/pull request creation ?😕

When we say contributing to open-source, it does not necesarilly mean that you need to know how to code. There are different ways in which you can contribute even if you are a non-coder – but having some coding skills will help you (and the projects) out a lot.

Some common contributions can be through:

- Adding a description to a project's documentation to elaborate on a certain point, mostly referred to as a README file 
- Giving guidance on a specific project and how to use it.
- Adding sample output to show how the code works.
- Writing in-depth tutorials for the project.

------------
## Create a new issue
If you spot a problem with the docs, search if an issue already exists. If a related issue doesn't exist, you can open a new issue using a relevant issue form.
Issues available for community contribution:
The following tags mark issues that are open for community contribution:
help wanted: Open to participation from the community but not necessarily beginner-friendly
good first issue: Open to participation from the community and friendly towards new contributors

The following tags mark issues that are not open for community contribution:
```🚧 status: blocked```: Blocked by other work that needs to be done first
```🧹 status```: ticket work required: Needs additional work before it is ready to be taken up
```🚦 status```: awaiting triage: Has not been triaged by a maintainer



-------------------

## Git Branching Naming
The Git branching model is powerful, but you need to manage the branches correctly and effectively.

1. Start branch name with a Group word
bug-logo-alignment-issue – the developer is trying to fix the logo alignment issue;
wip-ioc-container-added – the branch relates to the task to add an IoC container in progress.

2. Use Hyphen or Slash as Separators
eg - feature_upgrade

3.Git Branch with Author Name
Many companies prefer to add authors’ names into the branch names according to the format below:

```<author>_<branch-type>_<branch-name>```

4.Avoid using numbers only 
5.Avoid long descriptive names for long-lived branches

-------------------------------

## Pull Request
A clear, well-built PR is an essential part of a collaborative workflow, and through pairing
1.Give your feature branch a clear name
2.Give your commits and PRs active and descriptive titles
For both commit messages and PR titles, it's a good idea to use the present imperative tense — for example, use “Fix dashboard typo” instead of “Fixed” or “Fixes.”

## Others Practices 

- Using the command line:
Fork the repo so that you can make your changes without affecting the original project until you're ready to merge them.
- Always Use an Authorized Email to Commit Code and with Sign-off.
- Create a Meaningful .gitignore File.
- Add meaningful description in PR


[©iamrahul8](https://github.com/iamrahul8)

**Comment by iamrahul8** [comment]
# Challenge 24  
 Searching for good first issues to make contributions

Status :
Currently looking for issues that I am able to contribute to.
Will update soon...

**Comment by iamrahul8** [comment]
# Challenge 25
This is a continuation of the 24th Challenge
### Update
 PRs Merged⬇
![image](https://user-images.githubusercontent.com/72541381/192147449-c813cff1-d796-43b8-a9a2-19de1d002c84.png)
![image](https://user-images.githubusercontent.com/72541381/192147503-f0656d01-1cd5-450a-b96e-20476d0f4344.png)
![image](https://user-images.githubusercontent.com/72541381/192147522-809cb7ba-c8a2-41fd-ba11-0130e0876462.png)

**Comment by iamrahul8** [comment]
# Challenge 26 
Contributed to [Syknapse](https://github.com/Syknapse/Contribute-To-This-Project/pull/1673)
### PR merged ✌
![image](https://user-images.githubusercontent.com/72541381/192330769-5e109ee4-5454-45b6-8825-cd507471dd6f.png)

Small help posted in discord 😁
![image](https://user-images.githubusercontent.com/72541381/192333215-f11182a6-e743-4764-9498-d244fc8e6eff.png)

Will register for Hacktoberfest as soon as it will live [Link](https://hacktoberfest.com/?utm_medium=email&utm_source=event&utm_campaign=hacktoberfest_&utm_content=preptember-past-participants&mkt_tok=MTEzLURUTi0yNjYAAAGG0g0Q0_sgLm6yN27nyzcdhbE9Ia4TwXNCpEjZyTkVWXmzia3Cb2Hb20Y6aFWYjxbvu08xNjVYC2CQcwH24JfTk3wP68ArMDhJw9ifskKBSQ)

**Comment by kuraakhilesh8230** [comment]
congrats you have completed milestone-3 successfully and you are good to go🎉

**Comment by iamrahul8** [comment]
# Challenge 28✅


Projects where i contributed recently
- https://github.com/Syknapse/Contribute-To-This-Project

- https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems

- https://github.com/zero-to-mastery/start-here-guidelines

- https://github.com/iamrahul8/100daysofcode
### Recently Merged PRs

- https://github.com/Syknapse/Contribute-To-This-Project/pull/1673

- https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/pull/124

- https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/pull/119

- https://github.com/zero-to-mastery/start-here-guidelines/pull/16849#issue-1382201997

### Opened issue

- https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/issues/192
- https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/issues/193

### Created a Repo for beginners 

- [Good first Contribution](https://github.com/iamrahul8/make-your-first-contribution)

>Current status of contribution to open source @iamrahul8

**Comment by iamrahul8** [comment]
# Challenge 29 ✅
## Discussion created/idea pitched
- https://github.com/scaleracademy/open-source-projects/discussions/96
#### screenshot :
![image](https://user-images.githubusercontent.com/72541381/193047375-f79f978c-29f3-4b1a-bfff-70bdcc5a3da8.png)
---
## Some upvoted ideas⏫

![image](https://user-images.githubusercontent.com/72541381/193047726-f61c537b-3c8c-44d8-aa21-d73d61328bde.png)

- https://github.com/scaleracademy/open-source-projects/discussions/92#discussioncomment-3762796
![image](https://user-images.githubusercontent.com/72541381/193047812-92b47311-6509-4ff2-bac0-2ec0d3cc81a3.png)
- https://github.com/scaleracademy/open-source-projects/discussions/63#discussioncomment-3762597
![image](https://user-images.githubusercontent.com/72541381/193047873-7880f6a6-c38a-4a3e-b4ed-446f30c59854.png)

**Comment by iamrahul8** [comment]
# Challenge 30✅
Finally this is the last step of this entire [September Open Source Challenge](https://scalerdiscord.notion.site/scalerdiscord/Open-Source-September-Scaler-Discord-Community-f9306c7dc15a464ba1b6ea7ee30cdf0d) by Scaler 💙

![w8ZJLtJbmuph](https://user-images.githubusercontent.com/72541381/193263928-01298b90-0827-4b26-a685-50d69a54af8a.gif)
### My major takeaways : 
This challenge helps me in brushing my ```basics git skills```.  During the challenge phase, I made a few mistakes that helped me discover a few new ideas. Thanks a lot to @shivaylamba  for creating all of these challenges, from beginner to advanced.

I will inspire others to participate in open source projects and improve the world. Thank you @kuraakhilesh8230 for smooth checking of entire challenges steps. And everyone who helped me indirectly or directly whenever i need any help in this entire challenge. Last but not least, I want to thank ```Scaler Academy``` for organizing this fantastic challenge. I hope that some more this kind of task will be oranised in future.🙌

Thanks , Have a great day☺


[![forthebadge](https://forthebadge.com/images/badges/open-source.svg)](https://forthebadge.com)

**Comment by kuraakhilesh8230** [comment]
congrats you have completed milestone-4 successfully and you are good to go🎉
